test/jdk/javax/swing/JComboBox/DisabledComboBoxFontTestAuto.java

```diff
@@ -113,30 +114,45 @@ private static void testMethod() throws IOException {
                 eColor2 = new Color(enabledImage2.getRGB(x, y));
                 dColor1 = new Color(disabledImage.getRGB(x + 1, y));
                 dColor2 = new Color(disabledImage2.getRGB(x, y));
+            } else if (lafName.equals("Windows")) {
+                // In Windows LAF, the ComboBox sizes are different.
+                // Combo size is 116 x 22 where as Combo2 size is 115 X 20 and
+                // that results in pixel offset of 1px for width and 1px for height.
+                eColor1 = new Color(enabledImage.getRGB(x, y));
+                eColor2 = new Color(enabledImage2.getRGB(x + 1, y - 1));
+                dColor1 = new Color(disabledImage.getRGB(x, y));
+                dColor2 = new Color(disabledImage2.getRGB(x + 1, y - 1));
+            } else if (lafName.equals("GTK+")) {
+                // In GTK LAF, the ComboBox sizes are different.
+                // Combo size is 159 x 30 where as Combo2 size is 179 X 34 and
+                // that results in pixel offset of 10px for width and 2px for height.
+                eColor1 = new Color(enabledImage.getRGB(x, y));
+                eColor2 = new Color(enabledImage2.getRGB(x + 10, y + 2));
+                dColor1 = new Color(disabledImage.getRGB(x, y));
+                dColor2 = new Color(disabledImage2.getRGB(x + 10, y + 2));
             } else {
                 eColor1 = new Color(enabledImage.getRGB(x, y));
                 eColor2 = new Color(enabledImage2.getRGB(x, y));
                 dColor1 = new Color(disabledImage.getRGB(x, y));
                 dColor2 = new Color(disabledImage2.getRGB(x, y));
             }
             if ((!isColorMatching(eColor1, eColor2)) || (!isColorMatching(dColor1, dColor2))) {
-                isIdentical = false;
-                break;
+                failingLafs.append(lafName + ", ");
+                return;
             }
         }
-
-        if (isIdentical) {
-            System.out.println("PASSED");
-        } else {
-            failingLafs.append(lafName + ", ");
-        }
+        System.out.println("Test Passed: "+lafName);
     }
 
     private static boolean isColorMatching(Color c1, Color c2) {
-        if ((c1.getRed() != c2.getRed())
-                || (c1.getBlue() != c2.getBlue())
-                || (c1.getGreen() != c2.getGreen())) {
+        int redDiff = c1.getRed() - c2.getRed();
+        int blueDiff = c1.getBlue() - c2.getBlue();
+        int greenDiff = c1.getGreen() - c2.getGreen();
```

Comment:
Does this need Math.abs to get the absolute value of the difference?

Reply:
That makes sense. Updated.